torchrl/collectors/collectors.py

```diff
@@ -783,8 +782,9 @@ def _step_and_maybe_reset(self) -> None:
         done_or_terminated = (
             (done | truncated) if truncated is not None else done.clone()
         )
-
         if done_or_terminated.any():
+            traj_ids = self._tensordict.get(("collector", "traj_ids"))
+            traj_ids = traj_ids.clone()
```

Comment:
Why do we need this clone?

Reply:
these are integers so it's not expensive (not as expensive as others)
it's because down there we modify it in-place so it's either that or that.
If we don't clone, we will also modify the value we cache in rollout and all the values stacked on the output td will have the same content (which is bad)